### PR TITLE
14708 Payload trigger non string fix

### PR DESCRIPTION
**Pull request by Sergio-Mira**
https://swrvedev.jira.com/browse/SWRVE-14708

Original PR:
https://github.com/Swrve/swrve-ios-sdk/pull/180

### CODE REVIEW

File: SwrveSDK/SDK/Talk/SwrveTriggerCondition.m [hunk 1]
```diff
@@ -39,12 +39,19 @@ - (BOOL) hasFulfilledCondition:(NSDictionary *)payload {
...
     NSArray *payloadKeys = [payload allKeys];
-    
     if([payloadKeys containsObject:_key]) {
-        
         if([payload objectForKey:_key] != [NSNull null]) {
-            
-            NSString *payloadValue = [payload objectForKey:_key];
+            id payloadObject = [payload objectForKey:_key];
+            NSString *payloadValue = nil;
+            if ([payloadObject isKindOfClass:NSString.class]) {
```
Comment (adam-govan) [inline SwrveSDK/SDK/Talk/SwrveTriggerCondition.m]: `[NSString class]` is better than NSString.class

File: SwrveSDK/SDK/Talk/SwrveTriggerCondition.m [hunk 2]
```diff
@@ -39,12 +39,19 @@ - (BOOL) hasFulfilledCondition:(NSDictionary *)payload {
...
+                if ([payloadObject respondsToSelector:@selector(stringValue)]) {
```
Comment (adam-govan) [inline SwrveSDK/SDK/Talk/SwrveTriggerCondition.m:49]: ~~ if the value is a member of NSString class, why do we need an additional if statement here? we can parse it directly to NSString and should autobox all the other stuff AFAIK

File: SwrveSDK/SDK/Talk/SwrveTriggerCondition.m (hunk 1 above)
Comment (Sergio-Mira) [reply SwrveSDK/SDK/Talk/SwrveTriggerCondition.m]: Changed

File: SwrveSDK/SDK/Talk/SwrveTriggerCondition.m (hunk 2 above)
Comment (Sergio-Mira) [reply SwrveSDK/SDK/Talk/SwrveTriggerCondition.m:49]: I had introduced an id var but not reversed the if logic, done!

**Comment by adam-govan** [comment]
🚀  Great!

**Pull request merged**